compiler/src/main/java/bullet/impl/ComponentProcessingStep.java

```diff
@@ -114,35 +124,32 @@ private void generateObjectGraph(TypeElement element) {
           break;
         case SIMPLE_MEMBERS_INJECTION:
         case MEMBERS_INJECTOR:
-          membersInjectionMethods.add(methodDescriptor);
+          memberInjectorsTypeGraph.addVertex(methodDescriptor);
+          memberInjectionMethodsTypeDescriptorMap.put(methodDescriptor.type(), methodDescriptor);
+
+          for (ComponentMethodDescriptor pendingChild : potentialPendingEdges.get(methodDescriptor.type())) {
+            memberInjectorsTypeGraph.addEdge(pendingChild, methodDescriptor);
+          }
+
+          DeclaredType declaredType = (DeclaredType) methodDescriptor.type();
+          TypeElement typeElement = (TypeElement) declaredType.asElement();
+
+          List<TypeMirror> parents = new ArrayList<>(typeElement.getInterfaces());
+          parents.add(typeElement.getSuperclass());
+          for (TypeMirror parent : parents) {
+            ComponentMethodDescriptor parentDescriptor = memberInjectionMethodsTypeDescriptorMap.get(parent);
+            if (parentDescriptor == null) {
+              potentialPendingEdges.put(parent, methodDescriptor);
```

Comment:
Hmm, I need more time to digest and analyze that. E.g. what if in the `A implements I`, `B implements I2`, `I2 extends I` case, the component has member injection methods for `A`, `B` and `I`, but not `I2`: will `B` be correctly _connected_ to `I`? (or more simply, with `C extends B`, `B extends A` and member injection methods for `C` and `A` but not `B`, will `C` be _connected_ to `A`?)